### ISSUE TITLE
Introduce release workflow

### ISSUE THREAD

**Issue by widhalmt** [NETWAYS/ansible-collection-elasticstack#104]
Since I'm more an engineer and less a developer, I want to put ideas up for discussion. How about the following:

* As soon as we tag `1.0.0` we introduce a `support/1.x.y` branch. Whenever we add non breaking changes to the repository, we merge into that instead of main. If the merge succeeds, we merge this support branch into main
* When we introduce breaking changes, we merge directly to main
* When we see the time coming for a new major release, we start a `rc/2.x.y` branch. Non breaking changes from support will also be merged into this branch
* When all issues targeting the new release are closed, we tag `rc1` from the `rc` branch. We'll ask users to test this rc and report issues. When all these issues are fixed, we'll tag `rc2`and so forth. *Only* when there are no changes besides docs and comments, the last RC will be tagged as the new release
* We aim on supporting the current major release and the last major release
* The rest is documented as "semantic versioning"

**Comment by afeefghannam89** [comment]
Semantic versioning is the way to go and we're actually implementing it right now.

I think the rest adds unnecessary complexity for a team and a project this size. Managing separate branches for older releases and release candidates, as well as supporting two major versions at the same time, doesn't provide much value right now. These practices become important when we still need to maintain older releases or when users are actively testing pre-release versions, and we're not at that stage yet.

**Comment by afeefghannam89** [comment]
Current state, measured today:

* no git tags and no GitHub releases in this repository
* `galaxy.yml` says `version: 1.0.0`
* `netways.elasticstack` is not on Ansible Galaxy - the `netways` namespace there only holds
  `netways.icinga`
* `.github/workflows/` contains only test, lint, KICS and cache workflows, nothing for
  releasing or publishing

So the 1.0.0 milestone is currently unreachable in a practical sense: every issue in it could
be closed and there would still be nothing to ship.

What is left to do here:

* tag a release and create the matching GitHub release
* publish to Ansible Galaxy from that tag
* keep `galaxy.yml` in sync with the tag, following semantic versioning

The branching model proposed above - a `support/1.x.y` branch, `rc` branches, supporting two
major versions in parallel - stays out of scope for now, as discussed earlier in this issue.
Relabelled to `github_actions` and put into 1.0.0, since without a release path there is no
1.0.0.
